src/extension.ts

```diff
@@ -317,7 +331,7 @@ function launchMetals(
 
   context.subscriptions.push(client.start());
 
-  client.onReady().then(() => {
+  return client.onReady().then(() => {
```

Comment:
Added return to most functions so that both `withProgress` will work properly and the activasion function function will indeed finish at the proper time.